compiler/rustc_const_eval/src/interpret/terminator.rs

```diff
@@ -246,10 +244,28 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
     ) -> InterpResult<'tcx, Vec<FnArg<'tcx, M::Provenance>>> {
         ops.iter()
             .map(|op| {
-                Ok(match &op.node {
-                    mir::Operand::Move(place) => FnArg::InPlace(self.eval_place(*place)?),
-                    _ => FnArg::Copy(self.eval_operand(&op.node, None)?),
-                })
+                let arg = match &op.node {
+                    mir::Operand::Copy(_) | mir::Operand::Constant(_) => {
+                        let op = self.eval_operand(&op.node, None)?;
+                        FnArg::Copy(op)
+                    }
+                    mir::Operand::Move(place) => {
+                        let place = self.eval_place(*place)?;
+
+                        match place.as_mplace_or_local() {
```

Comment:
Ah right... this doesn't exclude the case where the local is in memory, we have to "normalize" to an `MPlaceTy`.

You can call `place_to_op` first and then `as_mplace_or_imm`.

Reply:
Yep. Not sure what exactly that does, but it fixes the problem, so 👍🏻